File: server/backend-app/src/main/java/com/sawwere/titlecounter/backend/app/storage/entity/Film.java
package com.sawwere.titlecounter.backend.app.storage.entity;

import jakarta.persistence.AttributeOverride;
import jakarta.persistence.AttributeOverrides;
import jakarta.persistence.Column;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;



@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "films")
public class Film {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(name = "title", nullable = false)
    private String title;

    @Column(name = "ru_title")
    private String ruTitle;

    @Column(name = "en_title")
    private String enTitle;

    @Embedded
    @AttributeOverrides({
            @AttributeOverride(name = "imdbId", column = @Column(name = "imdb_id", unique = true)),
            @AttributeOverride(name = "kpId", column = @Column(name = "kp_id", unique = true)),
            @AttributeOverride(name = "tmdbId", column = @Column(name = "tmdb_id", unique = true))
    })
    private FilmExternalId externalId;

    @Column(name = "description", length = 2048)
    private String description;

    @Column(name = "time")
    private Integer time;

    @Column(name = "year_release")
    private Integer yearRelease;

    @Column(name = "date_release")
    private LocalDate dateRelease;

    @Column(name = "global_score")
    private Float globalScore;

    @OneToMany(orphanRemoval = true, mappedBy = "film")
    private List<FilmEntry> filmEntries;

    @Column(name = "created_at", nullable = false)
    @ColumnDefault("'2024-08-04 10:23:54'::timestamp without time zone")
    @CreationTimestamp
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    @ColumnDefault("'2024-08-04 10:23:54'::timestamp without time zone")
    @UpdateTimestamp
    private LocalDateTime  updatedAt;
}

File: external-api-service/models/Film.py
class Film():
    def __init__(self,
                    movie, 
                    title = "ERROR NOT FOUND",
                    ru_title = None,
                    en_title = None,
                    image_url="https://kitairu.net/images/noimage.png",
                    kp_id=None,   
                    imdb_id=None, 
                    tmdb_id=None,
                    description=None,
                    time = 0,   
                    year_release = None,               
                    global_score = 0,
                    date_release = "1900-01-01"
                    ):
        self.movie = movie
        self.title = title
        self.ru_title = ru_title
        self.en_title = en_title
        self.image_url = image_url
        self.external_id = {"kp_id": kp_id, "imdb_id": imdb_id, "tmdb_id": tmdb_id}
        self.description = description
        self.time = time
        self.year_release = year_release
        self.date_release = date_release
        self.global_score = global_score
    
    def __getstate__(self):
        state = self.__dict__.copy()
        del state['movie']
        return state
    
    def to_dict(self) -> dict:
        return {"title": self.title,
                "ru_title" : self.ru_title,
                "en_title" : self.en_title,
                "image_url": self.image_url,
                "external_id": self.external_id,
                "description": self.description,
                "time": self.time,
                "year_release": self.year_release,
                "date_release": self.date_release,
                "global_score": self.global_score
                }

